config/main.py

```diff
@@ -132,14 +146,28 @@ def execute_systemctl(list_of_services, action):
             except SystemExit as e:
                 log_error("Failed to execute {} of service {} with error {}".format(action, service, e))
                 raise
+
         if (service + '.service' in generated_multi_instance_services):
-            for inst in range(num_asic):
-                try:
-                    click.echo("Executing {} of service {}@{}...".format(action, service, inst))
-                    run_command("systemctl {} {}@{}.service".format(action, service, inst))
-                except SystemExit as e:
-                    log_error("Failed to execute {} of service {}@{} with error {}".format(action, service, inst, e))
-                    raise
+            multi_inst_service_list.append(service)
+
+    # With Multi NPU, Start a thread per instance to do the "action" on multi instance services.
+    if sonic_device_util.is_multi_npu():
+        threads = []
+        # Use this event object to co-ordinate if any threads raised exception
+        e = threading.Event()
+        kwargs = {'multi_inst_list' : multi_inst_service_list, 'action' : action}
+        for inst in range(num_asic):
+            t = threading.Thread(target=execute_asic_instance, args=(inst,e), kwargs=kwargs)
```

Comment:
Ad space after comma in args tuple: `args=(inst, e)`

Reply:
Updated the code.